src/crush/mapper.c

```diff
@@ -73,6 +73,7 @@ int crush_find_rule(const struct crush_map *map, int ruleset, int type, int size
 static int bucket_perm_choose(struct crush_bucket *bucket,
 			      int x, int r)
 {
+       assert(bucket->size != 0);
```

Comment:
this isn't useful since we'd crash on the next line anyway and get the same stack trace.  but it does slow us down slightly.